pulsar-broker/src/main/java/org/apache/pulsar/broker/intercept/BrokerInterceptors.java

```diff
@@ -133,6 +168,18 @@ public void producerCreated(ServerCnx cnx, Producer producer,
         }
     }
 
+    @Override
+    public void producerClosed(ServerCnx cnx,
+                               Producer producer,
+                               Map<String, String> metadata) {
+        if (interceptors == null || interceptors.isEmpty()) {
+            return;
+        }
```

Comment:
I see many duplicated code here,
It's better to have a method that can be reused for all the method

```java
if (enabled) {
    for (BrokerInterceptorWithClassLoader value : interceptors.values()) {
            value.producerClosed(cnx, producer, metadata);
        }
}
```